RocketChat/Rocket.Chat.ReactNative
# [FIX] Same room called multiple times when jumping to message

## Proposed changes

Currently app pushes a new screen every time we navigate with a link to a message, which may lead to us having the same screens being called multiple times in the route. In order to solve this, instead of using `navigation.push` we use `navigation.replace`.
## Issue(s)	


## How to test or reproduce

In order to test this you should have 2 rooms, each should have message with a link that navigates to the other (e.g. Room 1 has link http://localhost:3000/group/channel-002?msg=ThisIsTheMsgId as a message, Room 2 has link http://localhost:3000/group/channel-001?msg=ThisIsAnotherMsgId as a message)

From room 1
1. Tap link to navigate to room 2
2. Tap link to navigate to room 1
3. Tap back button

It should take you to `RoomsListView`, since it didn't push new screens, it'll just replace current RoomView with the one we're navigating to.

## Screenshots
![Jun-21-2021 14-19-14](https://user-images.githubusercontent.com/36211892/122809212-ca41b600-d29b-11eb-8518-765d695a554c.gif)


## Types of changes




## Checklist



## Further comments


TODO: Add E2E test with this test case
